### PR TITLE
Create add_dataset.rst

### PR DESCRIPTION
**Pull request by Riyabelle25**
#### What does this implement/fix? Explain your changes.
Adds a section in the developers' guide for contributing new datasets to sktime

### CODE REVIEW

**Review by mloning — COMMENTED** [review]
Thanks @Riyabelle25, as discussed:

* add to `docs/source/developer_guide.rst`
* remove under-development warning from your page

File: docs/source/developer_guide/add_dataset.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Adding Datasets to sktime
+===========
+Follow these steps to add a new dataset to sktime: 
+
+*  Include CSV file or supported other format under :code:`sktime/datasets/data/<dataset-name>`
+*  Add :code:`load_<dataset-name>(...)` function in file :code:`sktime/datasets/base.py`
+*  Add :code:`<dataset-name>` to the list :code:`__all__ = [...` in file :code:`sktime/datasets/__init__.py`
+*  Add :code:`<dataset-name>` as argument to method :code:`included_datasets = (...` in file :code:`sktime/sktime/datasets/setup.py` 
+*  Add import statement :code:`from sktime.datasets.base import load_<dataset-name>(...)` in file :code:`sktime/sktime/datasets/setup.py`
```
Comment (RavenRudi) [inline docs/source/developer_guide/add_dataset.rst:11]: I think it should be in file `sktime/sktime/datasets/__init__.py`

Comment (Riyabelle25) [reply docs/source/developer_guide/add_dataset.rst:11]: Thanks for catching that one!

**Comment by mloning** [comment]
Now merged - thanks @Riyabelle25 🎉

**Pull request merged**